### PR TITLE
[RLlib] Make queue placement ops blocking

### PR DESCRIPTION
**Pull request by avnishn**
Signed-off-by: avnish <avnish@anyscale.com>

This change should fix issues with IMPALA and potentially APEX that stem from the various learner threads 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by avnishn** [comment]
We should probably go over the results. I'll upload them to tb dev and share them offline.

**Comment by avnishn** [comment]
The f-e release tests have a bug in them which is solved by @ArturNiederfahrenhorst 's pr: https://github.com/ray-project/ray/pull/26586

File: rllib/algorithms/apex_dqn/apex_dqn.py [hunk 1]
```diff
@@ -593,7 +593,7 @@ def wait_on_replay_actors() -> None:
                 # the minimum threshold yet.
                 if item:
                     self.learner_thread.inqueue.put(
-                        self.replay_sample_batches[0], timeout=0.001
+                        self.replay_sample_batches[0], block=True
```
Comment (gjoliver) [inline rllib/algorithms/apex_dqn/apex_dqn.py:596]: hey man, can you like write a lot of comments detailing the journey you went through and all the finding, so we know why we made the choice to block here, which not at all intuitive?
thanks.

Comment (avnishn) [reply rllib/algorithms/apex_dqn/apex_dqn.py:596]: yeah :)

**Comment by richardliaw** [comment]
Learning tests still failing?

**Review by gjoliver — APPROVED** [review]
thanks man

**Pull request merged**